src/main/java/com/google/idea/perf/CallTreeBuilder.kt

```diff
@@ -97,7 +101,9 @@ class CallTreeBuilder(
         val now = clock.sample()
         val stack = generateSequence(currentNode, Tree::parent).toList().asReversed()
         for (node in stack) {
-            node.wallTime += now - node.startWallTime
+            val elapsedTime = now - node.startWallTime
+            node.wallTime += elapsedTime
+            node.maxCallTime = node.maxCallTime.coerceAtLeast(elapsedTime)
             node.startWallTime = now
```

Comment:
Currently we set `node.startWallTime = now` when `buildAndReset()` is called.

This is fine for `wallTime` because it is accumulated along the way.  
However, it poses an issue for the `maxCallTime` calculation.

Reply:
Good catch! I fixed this issue in an upcoming commit, but I have one question. What would the root node's max wall time be while the call stack is not empty?

Reply:
I think there are a couple reasonable options

1. "max time" could consider calls still on the stack. So if `foo()` is still on the stack, and it has been on the stack for `400 ms`, then the max time for `foo()` should be at least `400 ms`.

2. Or, "max time" could be computed only from completed calls. In that case the "max time" for the root node would always be 0 (since it's always on the stack). That's OK because the root node is not a user-visible node anyway.

Either option is fine with me!

Reply:
Alright thanks! To stay consistent with the wall time's update rate, I decided to pick option 1. In this case, the root node will always increase to satisfy an invariant (it never pops), but it's more of an emergent property rather than a corner case.